dg/media/analytics/src/app/cards/cards.component.ts

```diff
@@ -1,67 +1,190 @@
+import { config } from '../../config';
 
 @Component({
   selector: 'app-cards',
   templateUrl: './cards.component.html',
   styleUrls: ['./cards.component.css']
 })
 
-export class CardsComponent implements OnInit {
+export class CardsComponent implements OnInit, AfterViewInit {
+
+  overallcharts = [];
+  recentcharts = [];
+  Dropdownitems = [];
   cardsOverall = [];
   cardsRecent = [];
 
-  constructor(private cardsService: CardsService, private sharedService: SharedService) {
-    this.sharedService.argsList$.subscribe(data => {
-      this.getData(data);
+  private charts = [];
+  private recentChartsData = {};
+  private chooseDateRange: string = '';
+  private cardGraphConfig = config.cardGraphConfig;
+  private cardsConfigs = config.cardsConfig;
+  generalConfigs = config.generalConfig;
+
+
+  constructor(private cardsService: CardsService, private _sharedService: SharedService
+    , private _globalfiltersharedService: GlobalFilterSharedService) {
+    this._globalfiltersharedService.argsList$.subscribe(data => {
+      let options = this.createParams();
+      this.getCardsData(options);
+    });
+    this._sharedService.argsList$.subscribe(filters => {
+      if (this.generalConfigs.applyFilterOnRecent) {
+        this.getCardsData(filters);
+      }
     });
   }
-  cardsConfigs = environment.cardsConfig;
+
   ngOnInit(): void {
     Object.keys(this.cardsConfigs).forEach(key => {
-      if (this.cardsConfigs[key].overall.show) {
-        this.cardsOverall.push({
-          'id': key,
-          'text': this.cardsConfigs[key].text
-        });
-      }
-      if (this.cardsConfigs[key].recent.show) {
-        this.cardsRecent.push({
-          'id': key,
-          'text': this.cardsConfigs[key].text
-        });
-      }
+      try {
+        if (this.cardsConfigs[key].overall.cards) {
+          this.initCards(this.cardsOverall, key);
+        } else if (this.cardsConfigs[key].overall.graph) {
+          this.initCardsGraph(this.overallcharts, key, 'overall');
+        }
+
+        if (this.cardsConfigs[key].recent.cards) {
+          this.initCards(this.cardsRecent, key);
+        }
+        else if (this.cardsConfigs[key].recent.graph) {
+          this.initCardsGraph(this.recentcharts, key, 'recent');
+        }
+      } catch (e) { }
     });
-    let options = {
-      webUrl: environment.url + "getData",
-      params: {
-        apply_filter: false,
-      }
+  }
+
+  initCards(cards, key): void {
+    let cardsobj: Cards = {
+      'id': key,
+      'text': this.cardsConfigs[key].text
     }
-    this.getData(options);
+    cards.push(cardsobj);
   }
 
-  public getData(options): any {
+  initCardsGraph(cards, data, option): void {
+
+    let cardsgraphobj = {} as CardsGraph;
+    cardsgraphobj.title = this.cardsConfigs[data].text;
+    cardsgraphobj.nativeChart = null;
+    cardsgraphobj.lastDataPoint = 0;
+
+    if (option == 'overall') {
+      cardsgraphobj.tagName = this.cardsConfigs[data].overall.text;
```

Comment:
if and else if block codes can be merged into one. As you are already calling initCardsGraph with overall and recent as string arguments. Just a suggestion, as it would look neet and better.

Reply:
some minor difference is there for different type of graph, but that could be done in a better way.